src/Controller/CRUDController.php

```diff
@@ -368,15 +368,12 @@ public function batchAction(Request $request)
         if ($data = json_decode($request->get('data', ''), true)) {
             $action = $data['action'];
             $idx = $data['idx'];
-            $allElements = $data['all_elements'];
+            $allElements = (bool) $data['all_elements'];
             $request->request->replace(array_merge($request->request->all(), $data));
         } else {
-            $request->request->set('idx', $request->get('idx', []));
-            $request->request->set('all_elements', $request->get('all_elements', false));
-
-            $action = $request->get('action');
-            $idx = $request->get('idx');
-            $allElements = $request->get('all_elements');
+            $action = $request->request->getAlnum('action');
```

Comment:
what does the getAlnum?

Reply:
Alpha numeric ?

Reply:
```php
    /**
     * Returns the alphabetic characters and digits of the parameter value.
     *
     * @param string $key     The parameter key
     * @param string $default The default value if the parameter key does not exist
     *
     * @return string The filtered value
     */
    public function getAlnum($key, $default = '')
    {
        return preg_replace('/[^[:alnum:]]/', '', $this->get($key, $default));
    }
```
> Returns the alphabetic characters and digits of the parameter value.

and returns a `string` value. Ref: https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/HttpFoundation/ParameterBag.php#L127